### PR TITLE
Introduce Mutiny

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: extensions/mailer/runtime/src/main/java/io/quarkus/mailer/runtime/MailTemplateInstanceImpl.java [hunk 1]
```diff
@@ -3,21 +3,18 @@
 import static io.quarkus.qute.api.VariantTemplate.SELECTED_VARIANT;
 import static io.quarkus.qute.api.VariantTemplate.VARIANTS;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.CompletableFuture;
+import java.util.*;
```
Comment (gastaldi) [inline extensions/mailer/runtime/src/main/java/io/quarkus/mailer/runtime/MailTemplateInstanceImpl.java]: Avoid star imports

File: extensions/mutiny/runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-mutiny-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>quarkus-mutiny</artifactId>
+    <name>Quarkus - Mutiny - Runtime</name>
```
Comment (gsmet) [inline extensions/mutiny/runtime/pom.xml]: We will need a description for all the new extension runtime artifacts.

File: extensions/resteasy-mutiny/deployment/src/test/java/io/quarkus/resteasy/mutiny/test/Async.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package io.quarkus.resteasy.mutiny.test;
+
+import java.lang.annotation.*;
```
Comment (gwenneg) [inline extensions/resteasy-mutiny/deployment/src/test/java/io/quarkus/resteasy/mutiny/test/Async.java]: This star import should be removed.

File: extensions/mailer/runtime/src/main/java/io/quarkus/mailer/ReactiveMailer.java [hunk 4]
```diff
@@ -4,8 +4,11 @@
...
  * A mailer to send email asynchronously.
+ * 
+ * @deprecated Use {@link io.quarkus.mailer.mutiny.ReactiveMailer}.
  */
-public interface ReactiveMailer {
+@Deprecated
+public interface ReactiveMailer extends io.quarkus.mailer.axle.ReactiveMailer {
 
     /**
      * Sends the passed emails.
```
Comment (gwenneg) [inline extensions/mailer/runtime/src/main/java/io/quarkus/mailer/ReactiveMailer.java]: The `CompletionStage<Void> send(Mail... mails);` method from this interface could be deleted as it is already defined in the super interface.

File: extensions/mutiny/deployment/src/main/java/io/quarkus/mutiny/deployment/MutinyProcessor.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package io.quarkus.mutiny.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+
+public class MutinyProcessor {
+
+    @BuildStep
+    public FeatureBuildItem registerFeature() {
+        return new FeatureBuildItem("mutiny");
```
Comment (gwenneg) [inline extensions/mutiny/deployment/src/main/java/io/quarkus/mutiny/deployment/MutinyProcessor.java]: The feature String is usually declared as a constant in `io.quarkus.deployment.builditem.FeatureBuildItem`

File: extensions/resteasy-mutiny/deployment/src/main/java/io/quarkus/resteasy/mutiny/deployment/ResteasyMutinyProcessor.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.resteasy.common.spi.ResteasyJaxrsProviderBuildItem;
+import io.quarkus.resteasy.mutiny.runtime.*;
+
+public class ResteasyMutinyProcessor {
+
+    @BuildStep
+    public FeatureBuildItem registerFeature() {
+        return new FeatureBuildItem("resteasy-mutiny");
```
Comment (gwenneg) [inline extensions/resteasy-mutiny/deployment/src/main/java/io/quarkus/resteasy/mutiny/deployment/ResteasyMutinyProcessor.java]: Same here.

**Comment by gsmet** [comment]
Closing as outdated.

**Pull request closed without merging**